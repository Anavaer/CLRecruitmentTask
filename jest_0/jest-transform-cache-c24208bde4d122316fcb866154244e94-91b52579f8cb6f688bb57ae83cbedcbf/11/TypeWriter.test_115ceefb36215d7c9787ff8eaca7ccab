98dc86f935e518d7f70af4d9afb44588
'use strict';var _react = require('react');var _react2 = _interopRequireDefault(_react);
var _TypeWriter = require('./TypeWriter');var _TypeWriter2 = _interopRequireDefault(_TypeWriter);
var _config = require('./config');
var _reactTestRenderer = require('react-test-renderer');var _reactTestRenderer2 = _interopRequireDefault(_reactTestRenderer);
var _enzyme = require('enzyme');
var _enzymeAdapterReact = require('enzyme-adapter-react-15');var _enzymeAdapterReact2 = _interopRequireDefault(_enzymeAdapterReact);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

(0, _enzyme.configure)({ adapter: new _enzymeAdapterReact2.default() });


it('renders correctly', function () {
  var component = _reactTestRenderer2.default.create(
  _react2.default.createElement(_TypeWriter2.default, {
    text: _config.textArray,
    interval: 120,
    delay: 1500,
    infinite: false }));


  expect(component).toMatchSnapshot();
});


test('Text rendered is Pierwszy kod do sukcesu', function () {
  var compenent = (0, _enzyme.shallow)(
  _react2.default.createElement(_TypeWriter2.default, {
    text: _config.textArray,
    interval: 120,
    delay: 1500,
    infinite: false }));


  var p = compenent.find('.typeWriter p');

  expect(p.text()).toEqual('Pierwszy kod do sukcesu');

});
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIlR5cGVXcml0ZXIudGVzdC5qcyJdLCJuYW1lcyI6WyJhZGFwdGVyIiwiaXQiLCJjb21wb25lbnQiLCJjcmVhdGUiLCJleHBlY3QiLCJ0b01hdGNoU25hcHNob3QiLCJ0ZXN0IiwiY29tcGVuZW50IiwicCIsImZpbmQiLCJ0ZXh0IiwidG9FcXVhbCJdLCJtYXBwaW5ncyI6ImFBQUEsOEI7QUFDQSwwQztBQUNBO0FBQ0Esd0Q7QUFDQTtBQUNBLDZEOztBQUVBLHVCQUFVLEVBQUVBLFNBQVMsa0NBQVgsRUFBVjs7O0FBR0FDLEdBQUcsbUJBQUgsRUFBd0IsWUFBTTtBQUM1QixNQUFNQyxZQUFZLDRCQUFTQyxNQUFUO0FBQ2hCO0FBQ0UsMkJBREY7QUFFRSxjQUFVLEdBRlo7QUFHRSxXQUFPLElBSFQ7QUFJRSxjQUFVLEtBSlosR0FEZ0IsQ0FBbEI7OztBQVFBQyxTQUFPRixTQUFQLEVBQWtCRyxlQUFsQjtBQUNELENBVkQ7OztBQWFBQyxLQUFLLDBDQUFMLEVBQWlELFlBQU07QUFDckQsTUFBTUMsWUFBWTtBQUNoQjtBQUNFLDJCQURGO0FBRUUsY0FBVSxHQUZaO0FBR0UsV0FBTyxJQUhUO0FBSUUsY0FBVSxLQUpaLEdBRGdCLENBQWxCOzs7QUFRQSxNQUFNQyxJQUFJRCxVQUFVRSxJQUFWLENBQWUsZUFBZixDQUFWOztBQUVBTCxTQUFPSSxFQUFFRSxJQUFGLEVBQVAsRUFBaUJDLE9BQWpCLENBQXlCLHlCQUF6Qjs7QUFFRCxDQWJEIiwiZmlsZSI6IlR5cGVXcml0ZXIudGVzdC5qcyIsInNvdXJjZXNDb250ZW50IjpbImltcG9ydCBSZWFjdCBmcm9tICdyZWFjdCc7XG5pbXBvcnQgVHlwZVdyaXRlciBmcm9tICcuL1R5cGVXcml0ZXInO1xuaW1wb3J0IHt0ZXh0QXJyYXl9IGZyb20gXCIuL2NvbmZpZ1wiO1xuaW1wb3J0IHJlbmRlcmVyIGZyb20gJ3JlYWN0LXRlc3QtcmVuZGVyZXInO1xuaW1wb3J0IHtzaGFsbG93LCBjb25maWd1cmV9IGZyb20gJ2VuenltZSc7XG5pbXBvcnQgQWRhcHRlciBmcm9tICdlbnp5bWUtYWRhcHRlci1yZWFjdC0xNSc7XG5cbmNvbmZpZ3VyZSh7IGFkYXB0ZXI6IG5ldyBBZGFwdGVyKCkgfSk7XG5cblxuaXQoJ3JlbmRlcnMgY29ycmVjdGx5JywgKCkgPT4ge1xuICBjb25zdCBjb21wb25lbnQgPSByZW5kZXJlci5jcmVhdGUoXG4gICAgPFR5cGVXcml0ZXJcbiAgICAgIHRleHQ9e3RleHRBcnJheX1cbiAgICAgIGludGVydmFsPXsxMjB9XG4gICAgICBkZWxheT17MTUwMH1cbiAgICAgIGluZmluaXRlPXtmYWxzZX1cbiAgICAvPlxuICApO1xuICBleHBlY3QoY29tcG9uZW50KS50b01hdGNoU25hcHNob3QoKTtcbn0pO1xuXG5cbnRlc3QoJ1RleHQgcmVuZGVyZWQgaXMgUGllcndzenkga29kIGRvIHN1a2Nlc3UnLCAoKSA9PiB7XG4gIGNvbnN0IGNvbXBlbmVudCA9IHNoYWxsb3coXG4gICAgPFR5cGVXcml0ZXJcbiAgICAgIHRleHQ9e3RleHRBcnJheX1cbiAgICAgIGludGVydmFsPXsxMjB9XG4gICAgICBkZWxheT17MTUwMH1cbiAgICAgIGluZmluaXRlPXtmYWxzZX1cbiAgICAvPik7XG5cbiAgY29uc3QgcCA9IGNvbXBlbmVudC5maW5kKCcudHlwZVdyaXRlciBwJyk7XG5cbiAgZXhwZWN0KHAudGV4dCgpKS50b0VxdWFsKCdQaWVyd3N6eSBrb2QgZG8gc3VrY2VzdScpO1xuXG59KTsiXX0=