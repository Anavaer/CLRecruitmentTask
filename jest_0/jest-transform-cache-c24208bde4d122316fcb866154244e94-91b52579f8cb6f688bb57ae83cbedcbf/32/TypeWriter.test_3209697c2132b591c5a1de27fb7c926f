a2dbf6ca4ff2f5bcd2465f28be23dc1a
'use strict';var _react = require('react');var _react2 = _interopRequireDefault(_react);
var _TypeWriter = require('./TypeWriter');var _TypeWriter2 = _interopRequireDefault(_TypeWriter);
var _config = require('./config');
var _reactTestRenderer = require('react-test-renderer');var _reactTestRenderer2 = _interopRequireDefault(_reactTestRenderer);
var _enzyme = require('enzyme');function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}


it('renders correctly', function () {
  var component = _reactTestRenderer2.default.create(
  _react2.default.createElement(_TypeWriter2.default, {
    text: _config.textArray,
    interval: 120,
    delay: 1500,
    infinite: false }));


  expect(component).toMatchSnapshot();
});


test('Text rendered is Pierwszy kod do sukcesu', function () {
  var compenent = (0, _enzyme.shallow)(
  _react2.default.createElement(_TypeWriter2.default, {
    text: _config.textArray,
    interval: 120,
    delay: 1500,
    infinite: false }));


  var p = compenent.find('.typeWriter p');

  expect(p.text()).toEqual('Pierwszy kod do sukcesu');

});
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIlR5cGVXcml0ZXIudGVzdC5qcyJdLCJuYW1lcyI6WyJpdCIsImNvbXBvbmVudCIsImNyZWF0ZSIsImV4cGVjdCIsInRvTWF0Y2hTbmFwc2hvdCIsInRlc3QiLCJjb21wZW5lbnQiLCJwIiwiZmluZCIsInRleHQiLCJ0b0VxdWFsIl0sIm1hcHBpbmdzIjoiYUFBQSw4QjtBQUNBLDBDO0FBQ0E7QUFDQSx3RDtBQUNBLGdDOzs7QUFHQUEsR0FBRyxtQkFBSCxFQUF3QixZQUFNO0FBQzVCLE1BQU1DLFlBQVksNEJBQVNDLE1BQVQ7QUFDaEI7QUFDRSwyQkFERjtBQUVFLGNBQVUsR0FGWjtBQUdFLFdBQU8sSUFIVDtBQUlFLGNBQVUsS0FKWixHQURnQixDQUFsQjs7O0FBUUFDLFNBQU9GLFNBQVAsRUFBa0JHLGVBQWxCO0FBQ0QsQ0FWRDs7O0FBYUFDLEtBQUssMENBQUwsRUFBaUQsWUFBTTtBQUNyRCxNQUFNQyxZQUFZO0FBQ2hCO0FBQ0UsMkJBREY7QUFFRSxjQUFVLEdBRlo7QUFHRSxXQUFPLElBSFQ7QUFJRSxjQUFVLEtBSlosR0FEZ0IsQ0FBbEI7OztBQVFBLE1BQU1DLElBQUlELFVBQVVFLElBQVYsQ0FBZSxlQUFmLENBQVY7O0FBRUFMLFNBQU9JLEVBQUVFLElBQUYsRUFBUCxFQUFpQkMsT0FBakIsQ0FBeUIseUJBQXpCOztBQUVELENBYkQiLCJmaWxlIjoiVHlwZVdyaXRlci50ZXN0LmpzIiwic291cmNlc0NvbnRlbnQiOlsiaW1wb3J0IFJlYWN0IGZyb20gJ3JlYWN0JztcbmltcG9ydCBUeXBlV3JpdGVyIGZyb20gJy4vVHlwZVdyaXRlcic7XG5pbXBvcnQge3RleHRBcnJheX0gZnJvbSBcIi4vY29uZmlnXCI7XG5pbXBvcnQgcmVuZGVyZXIgZnJvbSAncmVhY3QtdGVzdC1yZW5kZXJlcic7XG5pbXBvcnQge3NoYWxsb3d9IGZyb20gJ2VuenltZSc7XG5cblxuaXQoJ3JlbmRlcnMgY29ycmVjdGx5JywgKCkgPT4ge1xuICBjb25zdCBjb21wb25lbnQgPSByZW5kZXJlci5jcmVhdGUoXG4gICAgPFR5cGVXcml0ZXJcbiAgICAgIHRleHQ9e3RleHRBcnJheX1cbiAgICAgIGludGVydmFsPXsxMjB9XG4gICAgICBkZWxheT17MTUwMH1cbiAgICAgIGluZmluaXRlPXtmYWxzZX1cbiAgICAvPlxuICApO1xuICBleHBlY3QoY29tcG9uZW50KS50b01hdGNoU25hcHNob3QoKTtcbn0pO1xuXG5cbnRlc3QoJ1RleHQgcmVuZGVyZWQgaXMgUGllcndzenkga29kIGRvIHN1a2Nlc3UnLCAoKSA9PiB7XG4gIGNvbnN0IGNvbXBlbmVudCA9IHNoYWxsb3coXG4gICAgPFR5cGVXcml0ZXJcbiAgICAgIHRleHQ9e3RleHRBcnJheX1cbiAgICAgIGludGVydmFsPXsxMjB9XG4gICAgICBkZWxheT17MTUwMH1cbiAgICAgIGluZmluaXRlPXtmYWxzZX1cbiAgICAvPik7XG5cbiAgY29uc3QgcCA9IGNvbXBlbmVudC5maW5kKCcudHlwZVdyaXRlciBwJyk7XG5cbiAgZXhwZWN0KHAudGV4dCgpKS50b0VxdWFsKCdQaWVyd3N6eSBrb2QgZG8gc3VrY2VzdScpO1xuXG59KTsiXX0=