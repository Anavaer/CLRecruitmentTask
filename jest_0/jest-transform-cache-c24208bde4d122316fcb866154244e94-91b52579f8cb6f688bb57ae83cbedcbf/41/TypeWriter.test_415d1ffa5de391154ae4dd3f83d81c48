d97ed01dedf0f7a83131f25e06814e78
'use strict';var _react = require('react');var _react2 = _interopRequireDefault(_react);
var _TypeWriter = require('./TypeWriter');var _TypeWriter2 = _interopRequireDefault(_TypeWriter);
var _config = require('./config');
var _reactTestRenderer = require('react-test-renderer');var _reactTestRenderer2 = _interopRequireDefault(_reactTestRenderer);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}


it('renders correctly', function () {
  var component = _reactTestRenderer2.default.create(
  _react2.default.createElement(_TypeWriter2.default, {
    text: _config.textArray,
    interval: 120,
    delay: 1500,
    infinite: false }));



  expect(component).toMatchSnapshot();
});
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIlR5cGVXcml0ZXIudGVzdC5qcyJdLCJuYW1lcyI6WyJpdCIsImNvbXBvbmVudCIsImNyZWF0ZSIsImV4cGVjdCIsInRvTWF0Y2hTbmFwc2hvdCJdLCJtYXBwaW5ncyI6ImFBQUEsOEI7QUFDQSwwQztBQUNBO0FBQ0Esd0Q7OztBQUdBQSxHQUFHLG1CQUFILEVBQXdCLFlBQU07QUFDNUIsTUFBTUMsWUFBWSw0QkFBU0MsTUFBVDtBQUNoQjtBQUNFLDJCQURGO0FBRUUsY0FBVSxHQUZaO0FBR0UsV0FBTyxJQUhUO0FBSUUsY0FBVSxLQUpaLEdBRGdCLENBQWxCOzs7O0FBU0FDLFNBQU9GLFNBQVAsRUFBa0JHLGVBQWxCO0FBQ0QsQ0FYRCIsImZpbGUiOiJUeXBlV3JpdGVyLnRlc3QuanMiLCJzb3VyY2VzQ29udGVudCI6WyJpbXBvcnQgUmVhY3QgZnJvbSAncmVhY3QnO1xuaW1wb3J0IFR5cGVXcml0ZXIgZnJvbSAnLi9UeXBlV3JpdGVyJztcbmltcG9ydCB7dGV4dEFycmF5fSBmcm9tIFwiLi9jb25maWdcIjtcbmltcG9ydCByZW5kZXJlciBmcm9tICdyZWFjdC10ZXN0LXJlbmRlcmVyJztcblxuXG5pdCgncmVuZGVycyBjb3JyZWN0bHknLCAoKSA9PiB7XG4gIGNvbnN0IGNvbXBvbmVudCA9IHJlbmRlcmVyLmNyZWF0ZShcbiAgICA8VHlwZVdyaXRlclxuICAgICAgdGV4dD17dGV4dEFycmF5fVxuICAgICAgaW50ZXJ2YWw9ezEyMH1cbiAgICAgIGRlbGF5PXsxNTAwfVxuICAgICAgaW5maW5pdGU9e2ZhbHNlfVxuICAgIC8+XG4gICk7XG5cbiAgZXhwZWN0KGNvbXBvbmVudCkudG9NYXRjaFNuYXBzaG90KCk7XG59KTsiXX0=