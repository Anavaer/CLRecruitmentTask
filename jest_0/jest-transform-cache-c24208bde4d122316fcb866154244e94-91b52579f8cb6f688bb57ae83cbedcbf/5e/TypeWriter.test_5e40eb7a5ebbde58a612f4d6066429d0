6fd54797219aafed85346fd78e65dcb9
'use strict';var _react = require('react');var _react2 = _interopRequireDefault(_react);
var _TypeWriter = require('./TypeWriter');var _TypeWriter2 = _interopRequireDefault(_TypeWriter);
var _config = require('./config');
var _reactTestRenderer = require('react-test-renderer');var _reactTestRenderer2 = _interopRequireDefault(_reactTestRenderer);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

it('renders correctly', function () {
  var component = _reactTestRenderer2.default.create(
  _react2.default.createElement(_TypeWriter2.default, {
    text: _config.textArray,
    interval: 120,
    delay: 1500,
    infinite: false }));


  expect(component).toMatchSnapshot();
});
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIlR5cGVXcml0ZXIudGVzdC5qcyJdLCJuYW1lcyI6WyJpdCIsImNvbXBvbmVudCIsImNyZWF0ZSIsImV4cGVjdCIsInRvTWF0Y2hTbmFwc2hvdCJdLCJtYXBwaW5ncyI6ImFBQUEsOEI7QUFDQSwwQztBQUNBO0FBQ0Esd0Q7O0FBRUFBLEdBQUcsbUJBQUgsRUFBd0IsWUFBTTtBQUM1QixNQUFNQyxZQUFZLDRCQUFTQyxNQUFUO0FBQ2hCO0FBQ0UsMkJBREY7QUFFRSxjQUFVLEdBRlo7QUFHRSxXQUFPLElBSFQ7QUFJRSxjQUFVLEtBSlosR0FEZ0IsQ0FBbEI7OztBQVFBQyxTQUFPRixTQUFQLEVBQWtCRyxlQUFsQjtBQUNELENBVkQiLCJmaWxlIjoiVHlwZVdyaXRlci50ZXN0LmpzIiwic291cmNlc0NvbnRlbnQiOlsiaW1wb3J0IFJlYWN0IGZyb20gJ3JlYWN0JztcbmltcG9ydCBUeXBlV3JpdGVyIGZyb20gJy4vVHlwZVdyaXRlcic7XG5pbXBvcnQge3RleHRBcnJheX0gZnJvbSBcIi4vY29uZmlnXCI7XG5pbXBvcnQgcmVuZGVyZXIgZnJvbSAncmVhY3QtdGVzdC1yZW5kZXJlcic7XG5cbml0KCdyZW5kZXJzIGNvcnJlY3RseScsICgpID0+IHtcbiAgY29uc3QgY29tcG9uZW50ID0gcmVuZGVyZXIuY3JlYXRlKFxuICAgIDxUeXBlV3JpdGVyXG4gICAgICB0ZXh0PXt0ZXh0QXJyYXl9XG4gICAgICBpbnRlcnZhbD17MTIwfVxuICAgICAgZGVsYXk9ezE1MDB9XG4gICAgICBpbmZpbml0ZT17ZmFsc2V9XG4gICAgLz5cbiAgKTtcbiAgZXhwZWN0KGNvbXBvbmVudCkudG9NYXRjaFNuYXBzaG90KCk7XG59KTtcblxuXG4iXX0=