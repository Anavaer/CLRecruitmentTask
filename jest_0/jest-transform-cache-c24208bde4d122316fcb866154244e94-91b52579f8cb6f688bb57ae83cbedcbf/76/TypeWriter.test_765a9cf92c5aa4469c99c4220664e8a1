225b277d5cabbe840c70b510cf7f3ded
'use strict';var _react = require('react');var _react2 = _interopRequireDefault(_react);
var _TypeWriter = require('./TypeWriter');var _TypeWriter2 = _interopRequireDefault(_TypeWriter);
var _config = require('./config');
var _reactTestRenderer = require('react-test-renderer');var _reactTestRenderer2 = _interopRequireDefault(_reactTestRenderer);
var _enzyme = require('enzyme');
var _enzymeAdapterReact = require('enzyme-adapter-react-16');var _enzymeAdapterReact2 = _interopRequireDefault(_enzymeAdapterReact);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

(0, _enzyme.configure)({ adapter: new _enzymeAdapterReact2.default() });


it('renders correctly', function () {
  var component = _reactTestRenderer2.default.create(
  _react2.default.createElement(_TypeWriter2.default, {
    text: _config.textArray,
    interval: 120,
    delay: 1500,
    infinite: false }));


  expect(component).toMatchSnapshot();
});


test('Text rendered is Pierwszy kod do sukcesu', function () {
  var component = (0, _enzyme.shallow)(
  _react2.default.createElement(_TypeWriter2.default, {
    text: _config.textArray,
    interval: 120,
    delay: 1500,
    infinite: false }));


  var p = component.find('.typeWriter > p');

  expect(p.text()).toEqual('Pierwszy kod do sukcesu');

});
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIlR5cGVXcml0ZXIudGVzdC5qcyJdLCJuYW1lcyI6WyJhZGFwdGVyIiwiaXQiLCJjb21wb25lbnQiLCJjcmVhdGUiLCJleHBlY3QiLCJ0b01hdGNoU25hcHNob3QiLCJ0ZXN0IiwicCIsImZpbmQiLCJ0ZXh0IiwidG9FcXVhbCJdLCJtYXBwaW5ncyI6ImFBQUEsOEI7QUFDQSwwQztBQUNBO0FBQ0Esd0Q7QUFDQTtBQUNBLDZEOztBQUVBLHVCQUFVLEVBQUVBLFNBQVMsa0NBQVgsRUFBVjs7O0FBR0FDLEdBQUcsbUJBQUgsRUFBd0IsWUFBTTtBQUM1QixNQUFNQyxZQUFZLDRCQUFTQyxNQUFUO0FBQ2hCO0FBQ0UsMkJBREY7QUFFRSxjQUFVLEdBRlo7QUFHRSxXQUFPLElBSFQ7QUFJRSxjQUFVLEtBSlosR0FEZ0IsQ0FBbEI7OztBQVFBQyxTQUFPRixTQUFQLEVBQWtCRyxlQUFsQjtBQUNELENBVkQ7OztBQWFBQyxLQUFLLDBDQUFMLEVBQWlELFlBQU07QUFDckQsTUFBTUosWUFBWTtBQUNoQjtBQUNFLDJCQURGO0FBRUUsY0FBVSxHQUZaO0FBR0UsV0FBTyxJQUhUO0FBSUUsY0FBVSxLQUpaLEdBRGdCLENBQWxCOzs7QUFRQSxNQUFNSyxJQUFJTCxVQUFVTSxJQUFWLENBQWUsaUJBQWYsQ0FBVjs7QUFFQUosU0FBT0csRUFBRUUsSUFBRixFQUFQLEVBQWlCQyxPQUFqQixDQUF5Qix5QkFBekI7O0FBRUQsQ0FiRCIsImZpbGUiOiJUeXBlV3JpdGVyLnRlc3QuanMiLCJzb3VyY2VzQ29udGVudCI6WyJpbXBvcnQgUmVhY3QgZnJvbSAncmVhY3QnO1xuaW1wb3J0IFR5cGVXcml0ZXIgZnJvbSAnLi9UeXBlV3JpdGVyJztcbmltcG9ydCB7dGV4dEFycmF5fSBmcm9tIFwiLi9jb25maWdcIjtcbmltcG9ydCByZW5kZXJlciBmcm9tICdyZWFjdC10ZXN0LXJlbmRlcmVyJztcbmltcG9ydCB7c2hhbGxvdywgY29uZmlndXJlfSBmcm9tICdlbnp5bWUnO1xuaW1wb3J0IEFkYXB0ZXIgZnJvbSAnZW56eW1lLWFkYXB0ZXItcmVhY3QtMTYnO1xuXG5jb25maWd1cmUoeyBhZGFwdGVyOiBuZXcgQWRhcHRlcigpIH0pO1xuXG5cbml0KCdyZW5kZXJzIGNvcnJlY3RseScsICgpID0+IHtcbiAgY29uc3QgY29tcG9uZW50ID0gcmVuZGVyZXIuY3JlYXRlKFxuICAgIDxUeXBlV3JpdGVyXG4gICAgICB0ZXh0PXt0ZXh0QXJyYXl9XG4gICAgICBpbnRlcnZhbD17MTIwfVxuICAgICAgZGVsYXk9ezE1MDB9XG4gICAgICBpbmZpbml0ZT17ZmFsc2V9XG4gICAgLz5cbiAgKTtcbiAgZXhwZWN0KGNvbXBvbmVudCkudG9NYXRjaFNuYXBzaG90KCk7XG59KTtcblxuXG50ZXN0KCdUZXh0IHJlbmRlcmVkIGlzIFBpZXJ3c3p5IGtvZCBkbyBzdWtjZXN1JywgKCkgPT4ge1xuICBjb25zdCBjb21wb25lbnQgPSBzaGFsbG93KFxuICAgIDxUeXBlV3JpdGVyXG4gICAgICB0ZXh0PXt0ZXh0QXJyYXl9XG4gICAgICBpbnRlcnZhbD17MTIwfVxuICAgICAgZGVsYXk9ezE1MDB9XG4gICAgICBpbmZpbml0ZT17ZmFsc2V9XG4gICAgLz4pO1xuXG4gIGNvbnN0IHAgPSBjb21wb25lbnQuZmluZCgnLnR5cGVXcml0ZXIgPiBwJyk7XG5cbiAgZXhwZWN0KHAudGV4dCgpKS50b0VxdWFsKCdQaWVyd3N6eSBrb2QgZG8gc3VrY2VzdScpO1xuXG59KTsiXX0=