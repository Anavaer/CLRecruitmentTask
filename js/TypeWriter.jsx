import React from 'react';

class TypeWriter extends React.Component {

  constructor() {
    super();
    // Początkowa wartość state zawiera sposoby pokazywania elementów span
    // i zawartość wyświetlaną w paragrafach
    this.state = {
      firstLine: '',
      secondLine: '',
      firstSpan: 'inline',
      secondSpan: 'none',
    }
  }

  componentDidMount() {

    // Dokonuję destrukturyzacji obiektu props a także tworzę zmiennę
    // których będę używał do wyświetlania napisów
    const {text, interval, delay} = this.props;
    let textFirst = '';
    let textSecond = '';
    let index = 1;
    let newSentence = true;


    this.intervalFirst = setInterval((fullText = [...text[0]], typedText = textFirst) => {
      // Cały napis dodaje się litera po literze, zostaje zapisany w state i wyświetlony
      textFirst = typedText.concat(fullText[typedText.length]);
      this.setState({firstLine: textFirst});

      // Jeśli cały napis został już wyświetlony to:
      // Znika pierwszy span i pojawia się drugi
      // Pierwszy interwał zostaje wyczyszczony
      if (this.state.firstLine === text[0]) {
        this.setState({firstSpan: 'none', secondSpan: 'inline'});
        clearInterval(this.intervalFirst);

        // Zostaje odpalony drugi interwał literujący drugą linijkę tekstu
        this.intervalSecond = setInterval((fullText = [...text[index]], typedText = textSecond) => {

          // Przy każdym nowym zdaniu flaga newSentence odpala timeout trwający ilośc czasu podaną w props
          if (newSentence) {
            this.timeout = setTimeout(() => {
              newSentence = false;
            }, delay);
          } else {
            // Tak jak wcześniej zdanie zostaje wyświetlone litera po literze
            textSecond = typedText.concat(fullText[typedText.length]);
            this.setState({secondLine: textSecond});

            // Pod koniec każdego zdania index się zwiększa, wyświetlany tekst zeruję a flaga resetuje
            if (this.state.secondLine === text[index]) {
              index += 1;
              textSecond = '';
              newSentence = true;
            }

            // Pod koniec wyświetlania całości tablicy, w zależności of własności infinite podanej w props
            // następuje albo zakończenie wywietlania albo reset indexu do 1
            if (!this.props.infinite && index === text.length) {
              clearInterval(this.intervalSecond);
            } else if (index === text.length) {
              index = 1;
            }
          }

        }, interval)

      }
    }, interval);

  }

  componentWillUnmount() {
    clearInterval(this.intervalFirst);
  }

  render() {
    // Komponent renderuje diva z dwoma paragrafami a ich zawartość jest zależna od state
    // Po wygenerowaniu całego pierwszego napisu this.state.firstSpan zmienia się na none i
    // powoduje zniknięcie elementu span
    return (
      <div className="typeWriter">
        <p>{this.state.firstLine}<span style={{display: this.state.firstSpan}}>|</span></p>
        <p>{this.state.secondLine}<span style={{display: this.state.secondSpan}}>|</span></p>
      </div>
    );
  }
}

export default TypeWriter;