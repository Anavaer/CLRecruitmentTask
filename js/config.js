export const textArray = [
  'asdasdasd',
  'asdasdasd',
  'asdasdasd',
  'asdasdasd',
  'asdasdasd',
];