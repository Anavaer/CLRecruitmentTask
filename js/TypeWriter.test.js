import React from 'react';
import TypeWriter from './TypeWriter';
import {textArray} from "./config";
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const component = renderer.create(
    <TypeWriter
      text={textArray}
      interval={120}
      delay={1500}
      infinite={false}
    />
  );
  expect(component).toMatchSnapshot();
});


