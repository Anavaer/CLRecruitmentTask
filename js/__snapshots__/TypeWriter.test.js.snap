// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="typeWriter"
>
  <p>
    
    <span
      style={
        Object {
          "display": "inline",
        }
      }
    >
      |
    </span>
  </p>
  <p>
    
    <span
      style={
        Object {
          "display": "none",
        }
      }
    >
      |
    </span>
  </p>
</div>
`;
